doc/plugin_server_upstreamauthority_vault.md

```diff
@@ -0,0 +1,88 @@
+#  Upstream Authority "vault" Plugin
+
+The vault plugin signs intermediate CA certificates for SPIRE using the Vault PKI Engine.
+The plugin does not support the `PublishJWTKey` RPC and is therefore not appropriate for use in nested SPIRE topologies where JWT-SVIDs are in use.
+
+## Configuration
+
+The plugin accepts the following configuration options:
+
+| key | type | required | description | default |
+|:----|:-----|:---------|:------------|:--------|
+| vault_addr  | string |   | The URL of the Vault server. (e.g., https://vault.example.com:8443/) | `${VAULT_ADDR}` |
+| pki_mount_point  | string |  | Name of the mount point where PKI secret engine is mounted | pki |
+| ca_cert_path     | string |  | Path to a CA certificate file used to verify the Vault server certificate. Only PEM format is supported. | `${VAULT_CACERT}` |
+| insecure_skip_verify  | string |  | If true, vault client accepts any server certificates | false |
+| cert_auth        | struct |  | Configuration for the Client Certificate authentication method | |
+| token_auth       | struct |  | Configuration for the Token authentication method | |
+| approle_auth     | struct |  | Configuration for the AppRole authentication method | |
+
+The plugin supports **Client Certificate**, **Token** and **AppRole** authentication methods.
```

Comment:
Should we document any permissions required depending on the authentication method? E.g.: if a token is used, does the token need to be attached to a policy with certain capabilities?

Reply:
I agree with you. 
I'll try to improve the document.